### ISSUE TITLE
Strange alternating indentation of vertical keyword lists

### ISSUE THREAD

**Issue by equwal** [Malabarba/aggressive-indent-mode#122]
In Common Lisp, you often want to have a long vertical keyword list, such as in the `:export` portion of `defpackage`. This suggests an arrangement such as the following:
```lisp
(defpackage :example
  (:use :cl)
  (:export :thing1
           :thing2
           :thing3
```
However, aggressive-indent-mode chooses an indentation method that seems nonsensical (though I may be unaware of another case where this makes sense, I suspect it is accidental):
```lisp
(defpackage :example
  (:use :cl)
  (:export :thing1
   :thing2
           :thing3
    :thing4
           :thing5))
```
So we have this alternating behaviour. Of course I can do it like this:
```lisp
(defpackage :example
  (:use :cl)
  (:export
   :thing1
   :thing2
   :thing3))
```
though that isn't idiomatic. Thank you!

**Comment by Malabarba** [comment]
Hi there,

Actually, aggressive-indent doesn't really "choose" an indentation method.
It simply applies the indentation functions configured by the major mode.

If you disable aggressive-ident, and then select the whole region and call
M-x indent-region, do you get the same behaviour?

**Comment by equwal** [comment]
> If you disable aggressive-ident, and then select the whole region and call M-x indent-region, do you get the same behaviour?

Yes, I'll have to find out where lisp-mode's indentation is implemented and submit my issue there. Thank you.

**Comment by daninus14** [comment]
@equwal any updates on this??

**Comment by daninus14** [comment]
Actually, with indent-region once I removed aggressive-indent from my `init.el` it did indent correctly. @Malabarba any ideas?

**Comment by Malabarba** [comment]
Does CL mode have some kind of alternating indentation behaviour when you repeatedly press tab on one of those lines?
If so, that would explain the issue. 

Aggressive indent doesn't necessarily just call indent-region on the whole region. In some circumstances it might incrementally indent line-by-line, which can cause some weird behaviours like that if the major mode in question tries to be fancy with the indent-line function.

**Comment by daninus14** [comment]
I just tried it both with `emacs -Q` and with an emacs version where I am not requiring `aggressive-indent` and there was no alternating indentation behavior. Not only that, but some things were indented differently altogether.

I actually just selected all the file once I removed `aggressive-indent` and did `indent-region` and it solved all sorts of other issues like parameters to a function not aligning after the name of the function if placed on a new line, but rather just a couple of spaces to the right of the opening parenthesis.

Could it be that aggressive indent is applying a different indent strategy than whatever is the normal for CL?

I thought it could be due to this code I had in my configuration

```elisp
(when (locate-library "aggressive-indent")
  (add-hook 'lisp-mode-hook #'aggressive-indent-mode)
  (add-hook 'clojure-mode-hook #'aggressive-indent-mode)
  (add-hook 'scheme-mode-hook #'aggressive-indent-mode))
```

But I don't follow why that would do anything more than enable it.

Does aggressive-indent choose a strategy to indent or just applies the default `indent-region`/ `indent line by line`?

**Comment by Malabarba** [comment]
> Does aggressive-indent choose a strategy to indent or just applies the default `indent-region`/ `indent line by line`?

Depending on the situation, aggressive-indent might use different strategies. But they are all strategies defined by the major mode.
It might:
- call `indent-according-to-mode` on current line
- call `indent-region` on several lines at once
- call `indent-region` from point until end of line
